File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Queue<Integer> queue = new LinkedList<>();
        Queue<Integer> speedQueue = new LinkedList<>();
        
        for (int n : progresses) queue.offer(n);
        for (int n : speeds) speedQueue.offer(n);

        List<Integer> answer = new ArrayList<>();

        while (!queue.isEmpty()) {
            // 하루 진행
            int size = queue.size();
            for (int i = 0; i < size; i++) {
                int work = queue.poll();
                int speed = speedQueue.poll();
                queue.offer(work + speed);
                speedQueue.offer(speed);
            }

            // 배포 가능한 기능 세기
            int count = 0;
            while (!queue.isEmpty() && queue.peek() >= 100) {
                queue.poll();
                speedQueue.poll();
                count++;
            }

            if (count > 0) answer.add(count);
        }

        return answer.stream().mapToInt(i -> i).toArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const answer=[];
    const n=progresses.length;
    let index=0;
    while(index<n){
        const remain=Math.ceil((100-progresses[index])/speeds[index]);
        let cnt=1;
        index++;
        while(index<n){
            const num=Math.ceil((100-progresses[index])/speeds[index])
            if(num>remain) break;
            cnt++;
            index++;
        }
        answer.push(cnt);
    }
    return answer;
}